packages/CodingStandard/src/Fixer/Php/ClassStringToClassConstantFixer.php

```diff
@@ -0,0 +1,88 @@
+<?php declare(strict_types=1);
+
+namespace Symplify\CodingStandard\Fixer\Php;
+
+use Nette\Utils\Strings;
+use PhpCsFixer\Fixer\DefinedFixerInterface;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use SplFileInfo;
+
+final class ClassStringToClassConstantFixer implements DefinedFixerInterface
+{
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            '"SomeClass::class" references should be used over string.',
+            [
+                new CodeSample(
+'<?php      
+
+$className = "DateTime";  
+                '),
+                new CodeSample(
+'<?php      
+
+$interfaceName = "DateTimeInterface";  
+                '),
+            ]
+        );
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return true;
+    }
+
+    public function fix(SplFileInfo $file, Tokens $tokens): void
+    {
+        foreach (array_reverse($tokens->toArray(), true) as $index => $token) {
+            /** @var Token $token */
+            if (! $this->isStringToken($token)) {
+                continue;
+            }
+
+            $potentialClassOrInterfaceName = trim($token->getContent(), "'");
+            if (class_exists($potentialClassOrInterfaceName) || interface_exists($potentialClassOrInterfaceName)) {
```

Comment:
Class may not exist, because ECS is used for projects that uses own classes, so better is regexp checking. I am using this:

```php
			if (preg_match('#^[A-Z]\w*[a-z]\w*(\\\\[A-Z]\w*[a-z]\w*)*\z#', $potentialClassOrInterfaceName)) {
```